File: server/src/graphs/nodes/rag-node.js
// server/src/graphs/nodes/rag-node.js
import { ragChain } from '../../chains/rag-chain.js';

export const ragNode = async (state) => {
  const { userInput } = state;
  try {
    const result = await ragChain.invoke({ question: userInput });
    return { ragResult: result };
  } catch (err) {
    console.error('[ragNode]', err.message);
    return { ragResult: '查询知识库时出错' };
  }
};

File: server-py/app/graphs/nodes/rag_node.py
from app.chains.rag_chain import rag_chain


def rag_node(state):
    user_input = state["user_input"]
    try:
        result = rag_chain.invoke({"question": user_input})
        return {"rag_result": result}
    except Exception as err:
        print(f"[ragNode] {err}")
        return {"rag_result": "查询知识库时出错"}
